app/src/main/java/nl/mpcjanssen/simpletask/AddTask.kt

```diff
@@ -286,21 +286,22 @@ class AddTask : ThemedActionBarActivity() {
             return
         }
         log.info(TAG, "Saving ${enteredTasks.size} tasks, updating ${m_backup.size} tasks")
-        for ((i, task) in enteredTasks.withIndex()) {
-            if (m_backup.size > 0) {
-                val taskIndex = TodoList.pendingEdits.get(i).toInt()
-                TodoList.todoItems.get(taskIndex).update(task.text)
+        if (m_backup.size > 0) {
+            enteredTasks.forEachIndexed { i, task ->
```

Comment:
There is one possible functional difference here. Instead of doing the `m_backup.size` check for each task, we do it once at the start. I *think* the backup size should always be the same as the entered task size, so it shouldn't matter, but just wanted to point this out.